src/main/java/at/hannibal2/skyhanni/features/combat/endprotectortracker/EndstoneProtector.kt

```diff
@@ -0,0 +1,435 @@
+package at.hannibal2.skyhanni.features.combat.endprotectortracker
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.*
+import at.hannibal2.skyhanni.utils.*
+import at.hannibal2.skyhanni.utils.CollectionUtils.addAsSingletonList
+import at.hannibal2.skyhanni.utils.ConditionalUtils.afterChange
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import at.hannibal2.skyhanni.utils.RenderUtils.renderString
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.replaceAll
+import at.hannibal2.skyhanni.utils.TimeUtils.inWholeTicks
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import at.hannibal2.skyhanni.utils.tracker.SkyHanniTracker
+import at.hannibal2.skyhanni.utils.tracker.TrackerData
+import com.google.gson.annotations.Expose
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.client.Minecraft
+import net.minecraft.entity.item.EntityArmorStand
+import net.minecraft.init.Blocks
+import net.minecraft.util.ResourceLocation
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.DurationUnit
+
+object EndstoneProtector {
+
+    val config get() = SkyHanniMod.feature.combat.endstoneCounter
+    private var guiDisplay = ""
+    private var countdownStart: SimpleTimeMark = SimpleTimeMark.farPast()
+    private var shouldCountdown = false
+    val storage get() = ProfileStorageData.profileSpecific?.endstoneProtectorTracker
+    private var inNest = false
+    private var stage: Int = 0
```

Comment:
stage can be an enum, this will allow you to clean up the when statement for checking by having the y value in the enum. This is also better as right now it is just a magic number.